src/NUnitFramework/framework/Attributes/TestCaseSourceAttribute.cs

```diff
@@ -173,63 +173,51 @@ private IEnumerable<ITestCaseData> GetTestCasesFor(IMethodInfo method)
 
                         if (parms == null)
                         {
-                            // 3. An array was passed, it may be an object[]
-                            //    or possibly some other kind of array, which
-                            //    TestCaseSource can accept.
```

Comment:
Reading the code, it appears that you are trying to accept any array type, where we previously only accepted an array of objects to represent the arglist. Is that the case?